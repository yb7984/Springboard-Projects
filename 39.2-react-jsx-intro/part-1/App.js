const App = () => <div>
    <FirstComponent />
    <NamedComponent name="name" />
</div>
