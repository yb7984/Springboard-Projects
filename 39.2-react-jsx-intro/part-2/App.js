const App = () => <div>
    <Tweet username="u1" name="U1" date="2021-2-23" message="m1" />
    <Tweet username="u2" name="U2" date="2021-2-23" message="m2" />
    <Tweet username="u3" name="U3" date="2021-2-23" message="m3" />
</div>
