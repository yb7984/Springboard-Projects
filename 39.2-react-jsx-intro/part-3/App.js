const App = () => <div>
    <Person name="U1" age="10" hobbies={[]} />
    <Person name="U22222222" age="18" hobbies={["fishing" , "sail"]} />
    <Person name="U333333" age="23" hobbies={["soccer" , "basketball"]} />
</div>
