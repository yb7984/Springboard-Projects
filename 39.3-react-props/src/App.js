import Pokegame from "./Pokegame";

function App() {
  return <Pokegame />
}

export default App;
