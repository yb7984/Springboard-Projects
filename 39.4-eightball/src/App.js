import EightBall from "./EightBall";

function App() {
  return (
    <div>
      <EightBall />
    </div>
  );
}

export default App;
