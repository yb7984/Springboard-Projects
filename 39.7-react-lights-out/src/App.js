import React from "react";
import Board from "./Board";
import "./App.css";

/** Simple app that just shows the LightsOut game. */

function App() {
  return (
    <div className="App">
      <Board nrows={10} ncols={10} chanceLightStartsOn={0.9}/>
    </div>
  );
}

export default App;
