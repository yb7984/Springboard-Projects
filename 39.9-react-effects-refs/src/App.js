import './App.css';
import DeckOfCards from './DeckOfCards';

function App() {
  return (<DeckOfCards />);
}

export default App;
