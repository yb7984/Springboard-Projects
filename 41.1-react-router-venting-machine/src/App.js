import './App.css';
import VentingMachine from './VentingMachine';
import Soda from './Soda';
import Chips from './Chips';
import Candies from './Candies';
import { BrowserRouter, NavLink, Route } from 'react-router-dom';
import NavBar from './NavBar';

function App() {
  return (
    <div className="App">
      <BrowserRouter>
        <NavBar />
        <Route exact path="/"><VentingMachine /></Route>
        <Route exact path="/soda"><Soda /></Route>
        <Route exact path="/chips"><Chips /></Route>
        <Route exact path="/candies"><Candies /></Route>
      </BrowserRouter>
    </div>
  );
}

export default App;
