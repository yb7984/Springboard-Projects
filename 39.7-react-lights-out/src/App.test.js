import React from "react";
import { render } from "@testing-library/react";
import App from "./App";

//make thae random just out put 0.2 and 0.8 one after another
let current = 0.8;

Math.random = () => {
    if (current === 0.8) {
        current = 0.8;
        return 0.2;
    }
    current = 0.2;
    return 0.2;
}

it("renders without crashing", function () {
    render(<App />);
});

it("matches snapshot", function () {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
});