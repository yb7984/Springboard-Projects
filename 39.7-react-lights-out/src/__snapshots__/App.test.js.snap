// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <table
      class="Board"
    >
      <tbody>
        <tr>
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
        </tr>
        <tr>
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell "
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
        </tr>
        <tr>
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell "
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
        </tr>
        <tr>
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell "
            data-testid="cell"
          />
          <td
            class="Cell "
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell "
            data-testid="cell"
          />
        </tr>
        <tr>
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
        </tr>
        <tr>
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
        </tr>
        <tr>
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell "
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell "
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
        </tr>
        <tr>
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
        </tr>
        <tr>
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
        </tr>
        <tr>
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell "
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
          />
          <td
            class="Cell "
            data-testid="cell"
          />
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
